frontend/lib/util/clang-integration.cpp

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright 2021-2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/util/clang-integration.h"
+
+#include "chpl/framework/TemporaryFileResult.h"
+#include "chpl/framework/query-impl.h"
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/uast/ExternBlock.h"
+
+#include "../util/filesystem_help.h"
+
+#include "clang/Basic/TargetInfo.h"
```

Comment:
Compiling this with `CHPL_LLVM=none` fails for me locally with `fatal error: 'clang/Basic/TargetInfo.h' file not found`. I'm not sure why this doesn't happen for the LLVM none CI check on Linux. In any case these includes should probably be guarded with `HAVE_LLVM` macros.